build-tools/automation/azure-pipelines.yaml

```diff
@@ -440,23 +453,25 @@ stages:
 
     - template: yaml-templates/fail-on-issue.yaml
 
-  # Xamarin.Android (Smoke Tests MSBuild - Mac-0)
-  - template: yaml-templates/run-msbuild-mac-tests.yaml
+  # MSBuild Smoke Tests - Mac
+  - template: yaml-templates/run-msbuild-tests.yaml
     parameters:
-      job_name: mac_dotnet_tests_0
-      job_suffix: One .NET
-      nunit_categories: '| (TestCategory = SmokeTests $(DotNetNUnitCategories))'
-      target_framework: $(DotNetStableTargetFramework)
-      provisionatorChannel: ${{ parameters.provisionatorChannel }}
-
-  # Xamarin.Android (Smoke Tests MSBuild - Win-0)
-  - template: yaml-templates\run-msbuild-win-tests.yaml
+      testOS: macOS
+      jobName: mac_smoke_msbuild_tests
+      jobDisplayName: macOS > Tests > MSBuild
+      agentCount: 2
+      testFilter: cat = SmokeTests
+      installApkDiff: true
+
+  # MSBuild Smoke Tests - Win
+  - template: yaml-templates/run-msbuild-tests.yaml
     parameters:
-      job_name: win_dotnet_tests_0
-      job_suffix: One .NET
-      nunit_categories: '| (TestCategory = SmokeTests $(DotNetNUnitCategories))'
-      target_framework: $(DotNetStableTargetFramework)
-      provisionatorChannel: ${{ parameters.provisionatorChannel }}
+      testOS: Windows
+      jobName: win_smoke_msbuild_tests
+      jobDisplayName: Windows > Tests > MSBuild
+      agentCount: 2
+      testFilter: cat = SmokeTests
```

Comment:
Given our environment provisioning overhead, the new splitting approach, and recent reductions to classic tests -- I wonder if it is still beneficial to run MSBuild "SmokeTests" separately. I think we could simplify things by removing these jobs and the extra filtering we have to do, and even potentially save on overall machine time per PR. We could opt to do this in a later PR though.

Reply:
It does seem like the full MSBuild test suite is run on nearly all PR's, so it's rare for us to only run Smoke Tests and thus get any savings from skipping the full run.

I think this is a good exploration for a future PR, as this one is already pretty big.